gzip_logs_tools/log_compressor/log_compressor.c

```diff
@@ -790,7 +791,8 @@ create_pid_file(const char *pid_file)
 	}	
 
 	sprintf(buf, "%ld\n", (long) getpid());
-	if (write(fd, buf, strlen(buf)) != strlen(buf))
+	ssize_t written_bytes = write(fd, buf, strlen(buf));
+	if (written_bytes <0 || (size_t) written_bytes < strlen(buf)) // -1 if not written else size which is <= input_buffer)
```

Comment:
same as above, i'd cast the result of strlen to ssize_t. 
i think i copied this function from somewhere... but would be better to add a var - size_t buf_len and avoid calling strlen twice (it doesn't matter for performance here, but more correct)

Reply:
I don't think it is good form to cast strlen but I can live with that

Reply:
it's max 7 chars in this case, so... :)